### ISSUE TITLE
Rule proposal: `require-css-escape`

### ISSUE THREAD

**Issue by silverwind** [sindresorhus/eslint-plugin-unicorn#2318]
### Description

[CSS.escape](https://developer.mozilla.org/en-US/docs/Web/API/CSS/escape_static) should be used when interpolating variables into a CSS selector to ensure correct escaping.

### Fail

```js
document.querySelector(`#${id}`);
```

```js
el.querySelectorAll(`a[href^="#${hash}"]`);
```

### Pass

```js
document.querySelector(`#${CSS.escape(id)}`);
```

```js
document.querySelector(cssEscape`#${id}`);
```

```js
el.querySelectorAll(`a[href^="#${CSS.escape(hash)}"]`);
```

**Comment by sindresorhus** [comment]
Accepted

**Comment by silverwind** [comment]
To reduce code churn, maybe it should have an option to limit it to attribute selectors as indicated by `[` and `]` in the surrounding strings. Patterns like below are pretty common and not exactly a problem if the string contains only CSS-safe characters:

```js
const className = 'foo';
document.querySelectorAll(`.${classname}`);
```
